src/SixLabors.Fonts/TextOptions.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace SixLabors.Fonts
+{
+    /// <summary>
+    /// Provides configuration options for rendering text.
+    /// </summary>
+    public sealed class TextOptions
+    {
+        private float tabWidth = 4F;
+        private float dpi = 72F;
+        private float lineSpacing = 1F;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextOptions"/> class.
+        /// </summary>
+        /// <param name="font">The font.</param>
+        public TextOptions(Font font) => this.Font = font;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextOptions"/> class from properties
+        /// copied from the given instance.
+        /// </summary>
+        /// <param name="options">The options whose properties are copied into this instance.</param>
+        public TextOptions(TextOptions options)
+        {
+            this.Font = options.Font;
+            foreach (FontFamily family in options.FallbackFontFamilies)
+            {
+                this.FallbackFontFamilies.Add(family);
+            }
+
+            this.TabWidth = options.TabWidth;
+            this.ApplyHinting = options.ApplyHinting;
+            this.Dpi = options.Dpi;
+            this.LineSpacing = options.LineSpacing;
+            this.Origin = options.Origin;
+            this.WrappingLength = options.WrappingLength;
+            this.WordBreaking = options.WordBreaking;
+            this.TextDirection = options.TextDirection;
+            this.TextAlignment = options.TextAlignment;
+            this.HorizontalAlignment = options.HorizontalAlignment;
+            this.VerticalAlignment = options.VerticalAlignment;
+            this.LayoutMode = options.LayoutMode;
+            this.KerningMode = options.KerningMode;
+            this.ColorFontSupport = options.ColorFontSupport;
+        }
+
+        /// <summary>
+        /// Gets or sets the font.
+        /// </summary>
+        public Font Font { get; set; }
+
+        /// <summary>
+        /// Gets or sets the collection of fallback font families to use when
+        /// a specific glyph is missing from <see cref="Font"/>.
+        /// </summary>
+        public ICollection<FontFamily> FallbackFontFamilies { get; set; } = new HashSet<FontFamily>();
```

Comment:
I would avoid backing this with a heavy type like `HashSet<FontFamily>()`. How do we expect this to be consumed?
Can't it be just `IReadOnlyList<FontFamily>`, so it's `Array.Empty<FontFamily>` by default, and users can set it like:

```C#
textOptions.FallbackFontFamilies  = new FontFamily[] { ff1, ff2, ff3 };
```

Reply:
Yeah so I wanted to use `ICollection<FontFamily>` backed by `HashSet<FontFamily>` for two reasons.

1. I didn't want duplicate `FontFamily` in the collection. It's not harmful just wasteful. (I can probably deal with that in `TextLayout` though)
2. I wanted something we can immediately add to and continue to do so. However upon reflection that can be confusing since we can set the property itself. 

I'll update. You approach is better.

Reply:
Done